### ISSUE TITLE
Navigating form sections and focus state

### ISSUE THREAD

**Issue by jorisleker** [kiesraad/abacus#305]
- When navigating between form sections, we currently set the focus to the first input on the next page. 
- When there are unaccepted warnings or errors that prevent navigation, we scroll up the page, but we don't change the focus state. Focus remains were the user was before submitting the form section.

Both are not [recommended practice for single page applications](https://johnsweetaccessibility.com/2020/05/accessibility-in-spas-part-2/). 

What needs to change:
- [x] When **successfully** navigating to another form section, set focus to the h2 (this is the header that describes the content of the page, and corresponds to the menu-item clicked). 
- [x] When an error or unaccepted warning prevents a navigation attempt, set focus to the first feedback header 

nb: To be able to set focus to a non interactive element like an `h2` or `header`, add a `tabindex="-1"` attribute it (this will make it possible to focus the element, whilst ignoring it when tabbing through the page.
 
We might need something to indicate to the user that focus is on the first h2/feedback element. Will look into this after this ticket has been merged.

Something else that needs fixing:
- [x] The `ENTER` to move to another form field logic we build, prevents using `ENTER` from being used to confirm your focus on other elements. i.e.: when I focus an item in the progress list or a link in the header and press `ENTER` I expect to navigate to that step/page. Instead, the focus is moved to the 'next' form field. This should only happen when the focus is on another form field, not when it is somewhere else on the page.

**Comment by praseodym** [comment]
If we don't focus the first input field, users will first need to press the tab key to start data entry?

**Comment by jorisleker** [comment]
> If we don't focus the first input field, users will first need to press the tab key to start data entry?

Yes that's right. This is the way it should be done if we want to pass accessibility audits. Am open to discussion about this though, because we might unintentionally force more users to use their mouse to focus the first input on the page if we implement this. My goal would be to have as much users as possible use their keyboard only when entering numbers, because that's the more efficient way to do it. A strong focus indicator on the focussed heading might help here. 

Main issue that I want to tackle is outlined in the second bullet. If you submit a form section, and there are warnings/errors, something funny happens with the focus: it remains where it was (when using mouse to click 'volgende') or it is moved to the next input (if you use `SHIFT` + `ENTER` to submit). Both are confusing.
